packages/ember-runtime/lib/system/array_proxy.js

```diff
@@ -373,6 +373,17 @@ var ArrayProxy = EmberObject.extend(MutableArray, {
     this._super(...arguments);
     this._setupContent();
     this._setupArrangedContent();
+
+    if (!this._LEGACY_EMBER_DATA_SUPPORT) {
```

Comment:
Hm, this isn't better on `ENV` maybe? Ala `Ember.ENV._ENABLE_LEGACY_CONTROLLER_SUPPORT` ([usage](https://github.com/emberjs/ember.js/search?utf8=%E2%9C%93&q=_ENABLE_LEGACY_CONTROLLER_SUPPORT&type=Code))